# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.h
# Commit: 93f3b8aa5bc5
# Full Hash: 93f3b8aa5bc50a2e4d4e61c7ba0e839b681518a2
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-09 16:10:02
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.1: Untie callWithAbi from Frame::tls for Baseline. r=lth
#   
#   We have a tls dependency in callWithABI for wasm because we call builtins via BuiltinThunk.
#   Last one requires that tls to be loaded. In this patch I gradually extend callWithABI interface to
#   pass offset to tls. This allows us to preserve tls and load it by offset in callWithABI and frees us from
# ==============================================================================

diff -r 372f140d5b96 -r 93f3b8aa5bc5 js/src/jit/MacroAssembler.h
--- a/js/src/jit/MacroAssembler.h	Wed Sep 09 12:34:23 2020 +0000
+++ b/js/src/jit/MacroAssembler.h	Wed Sep 09 09:22:39 2020 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/EndianUtils.h"
 #include "mozilla/MacroForEach.h"
 #include "mozilla/MathAlgorithms.h"
+#include "mozilla/Maybe.h"
 
 #include "vm/Realm.h"
 
@@ -624,6 +625,7 @@
                           MoveOp::Type result = MoveOp::GENERAL);
 
   CodeOffset callWithABI(wasm::BytecodeOffset offset, wasm::SymbolicAddress fun,
+                         mozilla::Maybe<int32_t> tlsOffset,
                          MoveOp::Type result = MoveOp::GENERAL);
   void callDebugWithABI(wasm::SymbolicAddress fun,
                         MoveOp::Type result = MoveOp::GENERAL);